config/src/main/java/io/confluent/common/config/ConfigUtils.java

```diff
@@ -49,7 +49,10 @@ public static Properties translateDeprecated(Properties props, String[][] synony
     HashSet<String> synonymSet = new HashSet<>();
     for (String[] synonymGroup: synonymGroups) {
       for (String synonym : synonymGroup) {
-        synonymSet.add(synonym);
+        if (!synonymSet.add(synonym)) {
+          // TODO: we should consider doing more formal validations around this
+          log.warn("reused synonym={}", synonym);
```

Comment:
This is one of only a couple of lines not covered by tests. Is there something small and simple we could do to exercise it?